flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsException.java

```diff
@@ -39,12 +39,12 @@ public KinesisDataStreamsException(final String message, final Throwable cause)
 
         public KinesisDataStreamsFailFastException() {
             super(
-                    "Encountered an exception while persisting records, not retrying due to {failOnError} being set.");
+                    "Encountered an exception while persisting records, not retrying due to either: {failOnError} being set or the exception should not be retried.");
         }
 
         public KinesisDataStreamsFailFastException(final Throwable cause) {
             super(
-                    "Encountered an exception while persisting records, not retrying due to {failOnError} being set.",
+                    "Encountered an exception while persisting records, not retrying due to either: {failOnError} being set or the exception should not be retried.",
```

Comment:
Semantically it would make more sense to split the exception rather than adding ambiguity for user. For example, we could add a `KinesisDataStreamsFatalException`

Reply:
On second thoughts is this even correct? Seems like we are throwing `KinesisDataStreamsException` in this PR

Reply:
I'm currently wrapping any exception that I think is not retryable with the `KinesisDataStreamsException`, if we encounter it in the KDSSink.

I hear you, maybe it will be better to throw any `Error`s and `RuntimeException`s  directly without wrapping. This way these non-checked exceptions would be thrown without a KDSException wrapper confusing things - but obvs they must still be accepted by the consumer because even non-checked exceptions aren't fatal until they are consumed by the mailbox.

Also agree we can split failOnError and not retrying exceptions.

Reply:
Ah nevermind, I understand now.